core/src/main/java/com/graphhopper/util/GHUtility.java

```diff
@@ -168,8 +168,11 @@ private static void printUnitTestEdge(FlagEncoder encoder, EdgeIteratorState edg
                 "graph.edge(%d, %d, %f, %s); // edgeId=%s\n", from, to, edge.getDistance(), fwd && bwd ? "true" : "false", edge.getEdge());
     }
 
+    /**
+     * @param speed if null a random speed will be assign to every edge
+     */
     public static void buildRandomGraph(Graph graph, Random random, int numNodes, double meanDegree, boolean allowLoops,
-                                        boolean allowZeroDistance, DecimalEncodedValue randomSpeedEnc,
+                                        boolean allowZeroDistance, BooleanEncodedValue accessEnc, DecimalEncodedValue speedEnc, Double speed,
                                         double pNonZeroLoop, double pBothDir, double pRandomDistanceOffset) {
```

Comment:
Why did you add the `speed` parameter here? The previous behavior was that the default speed was used if `randomSpeedEnc` was null and random speeds were generated otherwise. In case `randomSpeedEnc.isStoreTwoDirections` different speeds were used for fwd/bwd direction. Ok now the default speed is gone we need some speed parameter (or just set it to 60 inside this method?), but we should still use random speeds were we did before. Right now speed is `!= null` almost everywhere. Also its a bit problematic to change this method because it might break older seeds (that we should not rely on, but I think in some cases we still do...)

Reply:
I did not intend to change this and tried to be careful ... likely I misunderstand something. The intention was to call with speed=null where before it was called with randomSpeedEnc=null. 

> Right now speed is != null almost everywhere.

Only for EdgeBasedTarjanSCCTest there were calls with randomSpeedEnc=null. Those two calls are done via speed=null now. 

> Also its a bit problematic to change this method because it might break older seeds

Hmmh, `random.nextDouble()` is still called regardless of whether it is used like before. For me currently it is not obvious where I changed behaviour. Probably we need a test for this :) ? I'll have a look as you observed behaviour change due to this in #1631

> but we should still use random speeds were we did before

I checked twice but if speed is null the speeds are random (which also feels more logical than before).

Reply:
> I did not intend to change this and tried to be careful ...

Yeah, I saw. I also checked and though its the same. But now I realized that once we set `speed != null` we always have `fwdSpeed==bwdSpeed`, while before we had `fwdSpeed!=bwdSpeed` in case `speedEnc!=null` :roll_eyes: ... Ugh I will have a look again tomorrow ;)